File: libs/legacy/feature-shell/src/lib/js/services/achievement/achievements-info.ts
export interface HsAchievementsInfo {
	readonly achievements: readonly HsAchievementInfo[];
}

export interface HsAchievementInfo {
	readonly id: number;
	readonly progress: number;
	readonly index?: number;
	readonly completed: boolean;
}

export interface HsAchievementCategory {
	readonly id: number;
	readonly icon: string;
	readonly name: string;
	readonly availablePoints: number;
	readonly points: number;
	readonly completedAchievements: number;
	readonly totalAchievements: number;
	readonly completionPercentage: number;
	readonly unclaimedAchievements: number;
}

File: tools/linux-helpers/unity-spy/UnitySpy.HearthstoneLib/Detail/Achievement/AchievementsInfo.cs
namespace HackF5.UnitySpy.HearthstoneLib.Detail.Achievement
{
    using System.Collections.Generic;
    
    public class AchievementsInfo : IAchievementsInfo
    {
        public IReadOnlyList<IAchievementInfo> Achievements { get; set; }
    }

    public class AchievementCategory
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Icon { get; set; }
        public AchievementCategoryStats Stats { get; set; }
    }

    public class AchievementCategoryStats
    {
        public int AvailablePoints { get; set; }
        public int Points { get; set; }
        public int CompletedAchievements { get; set; }
        public int TotalAchievements { get; set; }
        public int Unclaimed { get; set; }

    }
}